### PR TITLE
feat: `createAuth0OAuth2Client()`

### PR DESCRIPTION
**Pull request by zachauten**
I based it on the Okta provider, I believe they're pretty similar. Let me know if I missed anything!

Screen Shot 2023-07-15 at 4 45 16 PM

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -184,8 +185,8 @@ const client = new OAuth2Client({
   Twitter, the environment variable keys are `TWITTER_CLIENT_ID` and
   `TWITTER_CLIENT_SECRET`. See
   [the list below](#pre-configured-oauth-20-clients) for specifics.
-- `OKTA_DOMAIN` - required only when using the Okta provider to supply your
-  given Okta domain.
+- `OKTA_DOMAIN & AUTH0_DOMAIN` - required only when using the Okta or Auth0
```
Comment (iuioiua) [inline README.md:188]:
```suggestion
- `OKTA_DOMAIN` or `AUTH0_DOMAIN` - required only when using the Okta or Auth0
```

File: README.md [hunk 2]
```diff
@@ -184,8 +185,8 @@ const client = new OAuth2Client({
...
+  provider to supply your given Okta or Auth0 domain.
```
Comment (iuioiua) [inline README.md:189]:
```suggestion
  provider to supply your own given domain.
```

**Review by iuioiua — APPROVED** [review]
LGTM! Thank you.

**Comment by adoublef** [comment]
@zachauten I just wanted to ask if you are experiencing any issues with Auth0? I seem to be hitting this error `Error - OAuth2Strategy requires a clientID option` and it's weird as neither this nor the oauth_client package that it depends on seems to have changed anything recently.

**Comment by iuioiua** [comment]
Are you running your script with `AUTH0_CLIENT_ID=...` defined?

**Comment by adoublef** [comment]
```
signIn? {
  clientId: "Nr3U115Swf7ypKf*REDACTED*",
  clientSecret: "InbwTTVA9cxl3-*REDACTED*",
  authorizationEndpointUri: "https://*REDACTED*/oauth2/authorize",
  tokenUri: "https://*REDACTED*/oauth2/oauth/token",
  redirectUri: "http://localhost:8000/callback",
  defaults: { scope: "openid profile" }
}
callback? {
  clientId: "Nr3U115Swf7ypKf*REDACTED*",
  clientSecret: "InbwTTVA9cxl3-*REDACTED*",
  authorizationEndpointUri: "https://*REDACTED*/oauth2/authorize",
  tokenUri: "https://*REDACTED*/oauth2/oauth/token",
  redirectUri: "http://localhost:8000/callback",
  defaults: { scope: "openid profile" }
}
[uncaught application error]: Error - OAuth2Strategy requires a clientID option
```

These are the logs of the oauthClient during the `signin` and `callback` legs so I know it's being read from the env vars. Googling lead me to some people having issues with env's but I feel as though if that was the case it would have failed at the first leg surely?

**Comment by adoublef** [comment]
mhmmm I tried to use the demo code for oak and it was failing, I manually loaded the env for a set of different credentials and that seemed to have worked

**Comment by iuioiua** [comment]
Can you please open a new issue with all relevant details?

**Comment by adoublef** [comment]
I can but I don't think it's related to deno itself so opened a ticket with Auth0. I am able to use my teams tenant and applications perfectly. It's just when I use my own tenant it fails

**Pull request merged**